src/map/Map.tsx

```diff
@@ -19,22 +19,42 @@ type MapProps = {
 export default function({ selectedPath, paths, queryPoints, bbox, mapStyle }: MapProps) {
     const mapContainerRef: React.RefObject<HTMLDivElement> = useRef(null)
     const [map, setMap] = useState<Mapbox | null>(null)
+    const prevViewPort = useRef<ViewPort | null>(null)
```

Comment:
I wonder if the functionality to remeber the viewport when the map instance is replaced can be achieved with a functional state update https://reactjs.org/docs/hooks-reference.html#functional-updates 

Here we would get the previous ```map``` instance as parameter to ```useState``` for the map something like this maybe?
```
const mapWrapper = new Mapbox(
            mapContainerRef.current!,
            mapStyle,
            () => {
                setMap(prevWrapper => {
                    mapWrapper.setViewPort(prevWrapper.getViewPort())
                    return mapWrapper
                })
                Dispatcher.dispatch(new MapIsLoaded())
            }
        )
```
On the other hand this might be to late and cause fickering of the map again, because the viewport would be set after the map is loaded.

Reply:
Ah, the code example is for line 35, but it also concerns the useRef

Reply:
Ok interesting. I followed the pattern to use the previous state here: https://reactjs.org/docs/hooks-faq.html#how-to-get-the-previous-props-or-state. If you don't mind also here I would leave as is for now and either change this in a follow-up commit/PR or maybe we do not need to replace the map instance anyway using the react wrapper #87.

Reply:
Sorry, for the late answer. I totally agree especially, since we want to replace the map layer anyway.